### ISSUE TITLE
NextResult returns true after stored procedure

### ISSUE THREAD

**Issue by bgrainger** [mysql-net/MySqlConnector#135]
`MySqlDataReader.NextResultAsync` will return `true` after reading the results returned by the stored procedure. This appears to be because MySQL Server returns an empty resultset after the main results (https://dev.mysql.com/doc/internals/en/multi-resultset.html): 

> ... and a closing empty resultset, an OK_Packet. … The trailing OK_Packet is the response to the CALL statement and contains the affected-rows count of the last statement.

This will cause a crash with Dapper's QueryMultiple, because Dapper will think that the server returns an extra resultset that the client isn't reading:

```
connection.Execute(@"drop procedure if exists multiple_selects;
create procedure multiple_selects() begin select 1; select 2; end;");

using (var reader = connection.QueryMultiple("multiple_selects", commandType: CommandType.StoredProcedure))
{
	reader.Read<int>().Dump();
	reader.Read<int>().Dump();
} // <-- crash here, disposing the GridReader
```

**Comment by caleblloyd** [comment]
I think we can only handle this behavior when `commandType: CommandType.StoredProcedure`.  If the caller executes a `CommandType.Text` that contains `CALL multiple_selects`, we'd have to do a lot of query parsing to understand where we should remove the empty result set.

**Comment by bgrainger** [comment]
Agreed; it seems that if you're operating at that level, you need to be aware of how MySQL handles the `CALL` statement.

**Comment by caleblloyd** [comment]
I've got some WIP for this that involves creating a `MySqlResultSet` class and buffering `NextResult` for stored procedures.  Should have a PR open soon

**Comment by bgrainger** [comment]
Fixed in [0.7.1](https://www.nuget.org/packages/MySqlConnector/0.7.1).
